diff --git a/blueproxy.c b/blueproxy.c
index 9ecf6f1..83c33b0 100644
--- a/blueproxy.c
+++ b/blueproxy.c
@@ -1,3 +1,4 @@
+
 /*
  * Copyright (c) 2004 Anil Madhavapeddy <anil@recoil.org>
  * Copyright (c) 2004 Intel Research Cambridge
@@ -41,7 +42,7 @@
 #define DEFAULT_SOCKPATH "/tmp/bluespot"
 #define DEFAULT_PACKETSIZE 128
 #define MAX_PACKETSIZE 65535
-#define MAX_CHANNEL 16
+#define MAX_CHANNEL 32
 
 char hostname[1024] = "localhost";
 char search_name[256] = "Serial Port";
diff --git a/linux_bluez.c b/linux_bluez.c
index 330d2e7..a4dfac0 100644
--- a/linux_bluez.c
+++ b/linux_bluez.c
@@ -272,6 +272,20 @@ registerBlueproxyService(char *id, char *iface_name, int chan) {
     sdp_set_service_classes(&record, svclass_id);
     sdp_list_free(svclass_id, 0);
 
+#if 1
+    uuid_t svc_uuid;
+    uint32_t service_uuid_int[] = { 0x191A9851, 0x284A6259, 0xE12C0FAD, 0xA2164E65 }; // CouchDBReplication
+    sdp_uuid128_create(&svc_uuid, &service_uuid_int);
+
+    char str[256] = "";
+    sdp_uuid2strn(&svc_uuid, str, 256);
+    printf("Registering UUID %s\n", str);
+
+    svclass_id = sdp_list_append(0, &svc_uuid);
+    sdp_set_service_classes(&record, svclass_id);
+    sdp_list_free(svclass_id, 0);
+#endif
+
     sdp_uuid16_create(&profile.uuid, SERIAL_PORT_PROFILE_ID);
     profile.version = 0x0100;
     profiles = sdp_list_append(0, &profile);
